### ISSUE TITLE
Argument list from mc 1.13

### ISSUE THREAD

**Issue by Safranil** [tomsik68/mclauncher-api#36]
Since Minecraft 1.13, the arguments list provided in the JSON file has changed:

At 1.12, a string key `minecraftArguments` contain the arguments list to use.
<details>
<summary>JSON for 1.12.2</summary>

```json
{
  "minecraftArguments": "--username ${auth_player_name} --version ${version_name} --gameDir ${game_directory} --assetsDir ${assets_root} --assetIndex ${assets_index_name} --uuid ${auth_uuid} --accessToken ${auth_access_token} --userType ${user_type} --versionType ${version_type}"
}
```
</details>

Since 1.13, it's an array `arguments`. It's split JVM and Minecraft arguments and add some rules (for Win10, x86, ...)

<details>
<summary>JSON for 1.13</summary>

```json
{
  "arguments":{
    "game":[
      "--username",
      "${auth_player_name}",
      "--version",
      "${version_name}",
      "--gameDir",
      "${game_directory}",
      "--assetsDir",
      "${assets_root}",
      "--assetIndex",
      "${assets_index_name}",
      "--uuid",
      "${auth_uuid}",
      "--accessToken",
      "${auth_access_token}",
      "--userType",
      "${user_type}",
      "--versionType",
      "${version_type}",
      {
        "rules":[
          {
            "action":"allow",
            "features":{
              "is_demo_user":true
            }
          }
        ],
        "value":"--demo"
      },
      {
        "rules":[
          {
            "action":"allow",
            "features":{
              "has_custom_resolution":true
            }
          }
        ],
        "value":[
          "--width",
          "${resolution_width}",
          "--height",
          "${resolution_height}"
        ]
      }
    ],
    "jvm":[
      {
        "rules":[
          {
            "action":"allow",
            "os":{
              "name":"osx"
            }
          }
        ],
        "value":[
          "-XstartOnFirstThread"
        ]
      },
      {
        "rules":[
          {
            "action":"allow",
            "os":{
              "name":"windows"
            }
          }
        ],
        "value":"-XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump"
      },
      {
        "rules":[
          {
            "action":"allow",
            "os":{
              "name":"windows",
              "version":"^10\\."
            }
          }
        ],
        "value":[
          "-Dos.name=Windows 10",
          "-Dos.version=10.0"
        ]
      },
      {
        "rules":[
          {
            "action":"allow",
            "os":{
              "arch":"x86"
            }
          }
        ],
        "value":"-Xss1M"
      },
      "-Djava.library.path=${natives_directory}",
      "-Dminecraft.launcher.brand=${launcher_name}",
      "-Dminecraft.launcher.version=${launcher_version}",
      "-cp",
      "${classpath}"
    ]
  }
}
```
</details>

@tomsik68, do you think the class `sk.tomsik68.mclauncher.impl.versions.mcdownload.Rule` can do the trick?

Note: I can do the changes in a pull request to adapt the API.

See : [1.12.2](https://launchermeta.mojang.com/v1/packages/6e69e85d0f85f4f4b9e12dd99d102092a6e15918/1.12.2.json) and [1.13](https://launchermeta.mojang.com/v1/packages/1efd5dfcae060b847706500e408a76674922de89/1.13.json).

**Comment by tomsik68** [comment]
I think using the `Rule` class for this is correct. It might be worth to investigate the mysterious `features` field, but I don't think that's too important. Also, `MCDownloadVersionLauncher` will need to be modified accordingly.

Thank you for your dedication, your patch will be very welcome :smiley:

**Comment by Safranil** [comment]
No problem :)

Just a question, I started to look at `MCDownloadVersion`, Do you prefer to keep a `String` or can we change it to `List<String>` with a converter from `String` to `List` for backward compability ?

**Comment by tomsik68** [comment]
As the arguments never leave the package, feel free to change it to List. I would propose to add 2 lists into Version: `List<Argument> jvm, game;` where `Argument` is a class like this:
```java
final class Argument {
    private List<Rule> rules;
    private String value;
}
```
and then reflect the changes in launcher.

**Comment by tomsik68** [comment]
Now fixed in dce331753c259e55fef4c5685a5984547a3d5b8c
